test/new-e2e/tests/installer/upgrade_scenario_test.go

```diff
@@ -118,6 +116,57 @@ func (s *upgradeScenarioSuite) TestBackendFailure() {
 	s.assertSuccessfulStopExperiment(timestamp)
 }
 
+func (s *upgradeScenarioSuite) TestExperimentFailure() {
+	s.RunInstallScript("DD_REMOTE_UPDATES=true")
+	defer s.Purge()
+	s.host.WaitForUnitActive(
+		"datadog-agent.service",
+		"datadog-agent-trace.service",
+		"datadog-agent-process.service",
+		"datadog-installer.service",
+	)
+
+	s.setCatalog(testCatalog)
+
+	// Also tests if the version is not available in the catalog
+	_, err := s.startExperimentCommand(unknownAgentImageVersion)
+	require.Error(s.T(), err)
+
+	// Receive a failure from the experiment, stops the experiment
+	beforeStatus := s.getInstallerStatus()["datadog-agent"]
+	_, err = s.stopExperimentCommand()
+	require.NoError(s.T(), err)
+	afterStatus := s.getInstallerStatus()["datadog-agent"]
+
+	require.Equal(s.T(), beforeStatus, afterStatus)
+}
+
+func (s *upgradeScenarioSuite) TestExperimentCurrentVersion() {
+	s.RunInstallScript("DD_REMOTE_UPDATES=true")
+	defer s.Purge()
+	s.host.WaitForUnitActive(
+		"datadog-agent.service",
+		"datadog-agent-trace.service",
+		"datadog-agent-process.service",
+		"datadog-installer.service",
+	)
+
+	currentVersion := s.getInstallerStatus()["datadog-agent"].StableVersion
+	newCatalog := catalog{
+		Packages: []packageEntry{
+			{
+				Package: "datadog-agent",
+				Version: currentVersion,
+				URL:     fmt.Sprintf("oci://gcr.io/datadoghq/agent-package:%s", currentVersion),
+			},
+		},
+	}
+
+	s.setCatalog(newCatalog)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏Should be the first step